docs/Testing.md

```diff
@@ -110,7 +110,10 @@ tap.test('GET `/` route', t => {
 ```
 
 ### Testing with a running server
-Fastify can also be tested after starting the server with `fastify.listen()` or after initializing routes and plugins with `fastify.ready()`.
+Fastify can also be tested after starting the server with `fastify.listen()` or after initializing routes and plugins with `fastify.ready()`. 
+
+Note: `fastify.inject`, `fastify.listen`, or `fastify.ready` must be invoked before the callback of `fastify.register(plugin).after(callback)` will execute.
```

Comment:
What do you mean exactly? I’m not sure this is clear. Maybe can you add an example as well?

Reply:
@mcollina I have an example here: [issue #146 on fastify/help](https://github.com/fastify/help/issues/146).

I was basically trying to test a plugin but never invoked `fastify.ready` `fastify.inject` or `fastify.listen` within my tests. Because of this, `fastify.register(myFP).after` was never firing off.

Reply:
The way I solved this moving forward was to create a harness that looks like this: 

This example uses TypeScript and jest. 
```js
// helpers.ts
import core from '../src/app';
import Fastify = require('fastify');

let app;

export const build = (done: jest.DoneCallback): Fastify.FastifyInstance => {
  app = Fastify().register(core);
  app.ready(done);
  return app;
};

export const teardown = () => {
  app.close();
};
```

So my tests look like this: 

```js
import expect from 'expect';
import { build, teardown } from '../helper';
import { FastifyInstance } from 'fastify';

let app: FastifyInstance;

beforeAll(cb => {
  app = build(cb);
});
afterAll(() => {
  teardown();
});
describe('JWT Plugin', () => {
  it(`can sign and verify JWT`, done => {
    expect(testCondition)
  })
})
```

As a silly aside, I went from tap -> mocha -> jest thinking that the testing framework may have been somehow responsible. Woops. I was running into performance issues with tap anyway.

Reply:
On reflection, my usecase may be a bit different than what's typical. I have mine structured so that the "app" is exported as a fastify-plugin. My entry point then registers the app and starts the server. The reason for this is so that I can easily break up my app into microservices.

My tests have to do the same, which is where I was running into the problem. I was trying to use `after` to determine that the app was ready for testing without calling `inject`, `ready` or `listen`.